src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -3284,71 +3309,392 @@ new_readdir(DIR *dirp)
- *		num_entry = 0;
- *	} else if (envp[0] == NULL) {
- *		num_entry = 0;
- *	} else {
- *		while (envp[num_entry])	{
- *			num_entry++;
- *		}
- *	}
- *
- *	new_envp = malloc(sizeof(char *) * (num_entry + 5));
- *	assert(new_envp != NULL);
- *	for (i = 0; i < num_entry; i++)	{
- *		new_envp[i] = envp[i];
- *	}
- *	for (i = 0; i < 5; i++)	{
- *		new_envp[num_entry + i] = envp_lib[i];
- *	}
- *
- *	return new_envp;
- *}
- *
- *static int
- *new_execve(const char *filename, char *const argv[], char *const envp[])
- *{
- *	printf("DBG> execve(%s)\n", filename);
- *	return real_execve(filename, argv, pre_envp(envp));
- *}
- *
- *static int
- *new_execvp(const char *filename, char *const argv[])
- *{
- *	printf("DBG> execvp(%s)\n", filename);
- *	return real_execve(filename, argv, pre_envp(NULL));
- *}
- *
- *static int
- *new_execv(const char *filename, char *const argv[])
- *{
- *	printf("DBG> execv(%s)\n", filename);
- *	pre_envp(NULL);
- *	return real_execve(filename, argv, pre_envp(NULL));
- *}
- *
- *static pid_t
- *new_fork(void)
- *{
- *	pid_t pid;
- *	pid = real_fork();
- *	if (pid) {
- *		// parent process: do nothing
- *		return pid;
- *	} else {
- *		init_dfs();
- *		return pid;
- *	}
- *}
+ * char    env_list[N_ENV_CHECK][32] = {"LD_PRELOAD", "D_IL_REPORT", "DAOS_MOUNT_POINT",
+ *				     "DAOS_POOL", "DAOS_CONTAINER", "D_IL_MAX_EQ",
+ *				     "D_IL_ENFORCE_EXEC_ENV"};
  */
 
+static char **
+pre_envp(char *const envp[])
+{
+	int    i, rc, num_entry = 0;
+	int    idx_preload = -1;
+	int    len, len2, len_total;
+	char **new_envp;
+	char  *pil4df_path;
+	bool   preload_included     = false;
+	bool   pil4dfs_in_preload   = false;
+	bool   report_included      = false;
+	bool   mp_included          = false;
+	bool   pool_included        = false;
+	bool   cont_included        = false;
+	bool   maxeq_included       = false;
+	bool   enforcement_included = false;
+	char  *fs_root              = NULL;
+	char  *pool                 = NULL;
+	char  *container            = NULL;
+	char  *new_preload_str      = NULL;
+	char  *str_report           = NULL;
+	char  *str_mp               = NULL;
+	char  *str_pool             = NULL;
+	char  *str_cont             = NULL;
+	char  *str_maxeq            = NULL;
+	char  *str_enforcement      = NULL;
+
+	if (envp == NULL) {
+		num_entry = 0;
+	} else if (envp[0] == NULL) {
+		num_entry = 0;
+	} else {
+		while (envp[num_entry]) {
+			if (strncmp(envp[num_entry], "LD_PRELOAD", 10) == 0) {
+				preload_included = true;
+				idx_preload      = num_entry;
+				if (strstr(envp[num_entry], "libpil4dfs.so"))
+					pil4dfs_in_preload = true;
+			} else if (strncmp(envp[num_entry], "D_IL_REPORT", 11) == 0) {
+				report_included = true;
+			} else if (strncmp(envp[num_entry], "DAOS_MOUNT_POINT", 16) == 0) {
+				mp_included = true;
+			} else if (strncmp(envp[num_entry], "DAOS_POOL", 9) == 0) {
+				pool_included = true;
+			} else if (strncmp(envp[num_entry], "DAOS_CONTAINER", 14) == 0) {
+				cont_included = true;
```

Comment:
what about other env variables that DAOS users might set? 
D_LOG_MASK, D_LOG_FILE, etc ?